examples/mnist/demo/demo.go

```diff
@@ -0,0 +1,66 @@
+/*
+ *	Copyright 2025 Rener Castro
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License");
+ *	you may not use this file except in compliance with the License.
+ *	You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software
+ *	distributed under the License is distributed on an "AS IS" BASIS,
+ *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *	See the License for the specific language governing permissions and
+ *	limitations under the License.
+ */
+
+// demo for mnist library
+//
+//  1. With `demo --download`: it will simply download and unpack dataset.
+//  2. With `demo --train`: trains a CNN (convolutional neural network) model for "MNIST".
+package main
+
+import (
+	"flag"
+
+	"github.com/gomlx/exceptions"
+	"github.com/gomlx/gomlx/examples/mnist"
+	"github.com/gomlx/gomlx/ml/context"
+	"github.com/janpfeifer/must"
+	"k8s.io/klog/v2"
+
+	_ "github.com/gomlx/gomlx/backends/xla"
+)
+
+var (
+	flagTrain    = flag.Bool("train", true, "Flag to train")
+	flagDownload = flag.Bool("download", false, "Flag to download")
+	flagModel    = flag.String("model", "logistic", "Model function")
```

Comment:
"logistic" is usually associated with the loss.

A model is usually called "linear" or "cnn" or "fnn" or "kan" , etc. What I mean is, it's usually qualified by the type of function it's learning (linear, cnn, fnn, kan being different type of mathematical functions that model stuff).